packages/celotool/src/e2e-tests/celo_dollar_gas_currency_tests.ts

```diff
@@ -0,0 +1,654 @@
+    }
+    await initAndStartGeth(hooks.gethBinaryPath, fullInstance)
+
+    // Give the node time to sync the latest block.
+    await sleep(10)
+
+    await hooks.after()
+  }
+
+  const unlockAccount = async (address: string) => {
+    // Assuming empty password
+    await web3.eth.personal.unlockAccount(address, '', 1000)
+  }
+
+  const transferCeloGold = async (
+    fromAddress: string,
+    toAddress: string,
+    amount: BigNumber,
+    txOptions: any = {},
+    event: any = 'confirmation'
+  ) => {
+    await unlockAccount(fromAddress)
+    const tx: Tx = {
+      from: fromAddress,
+      to: toAddress,
+      value: amount.toString(),
+      ...txOptions,
+    }
+    if (!tx.gas) {
+      tx.gas = await web3.eth.estimateGas(tx)
+    }
+
+    return new Promise(async (resolve, reject) => {
+      try {
+        if (event === 'transactionHash') {
+          await web3.eth.sendTransaction(tx).on(event, (hash: any) => {
+            resolve(hash)
+          })
+        } else if (event === 'confirmation') {
+          await web3.eth.sendTransaction(tx).on(event, (_: any, receipt: any) => {
+            resolve(receipt)
+          })
+        }
+      } catch (err) {
+        reject(err)
+      }
+    })
+  }
+
+  const transferCeloDollars = async (
+    fromAddress: string,
+    toAddress: string,
+    amount: BigNumber,
+    txOptions: any = {}
+  ) => {
+    await unlockAccount(fromAddress)
+    // Hack to get the node to suggest a price for us.
+    // Otherwise, web3 will suggest the default gold price.
+    if (txOptions.gasCurrency) {
+      txOptions.gasPrice = '0'
+    }
+    const tx = stableToken.methods.transfer(toAddress, amount.toString())
+    let gas = txOptions.gas
+    if (!gas) {
+      gas = await tx.estimateGas({ ...txOptions })
+    }
+
+    return new Promise(async (resolve, reject) => {
+      try {
+        await tx
+          .send({ from: fromAddress, ...txOptions, gas })
+          .on('confirmation', (_: any, receipt: any) => resolve(receipt))
+      } catch (err) {
+        reject(err)
+      }
+    })
+  }
+
+  const convertCGToCD = (cGold: BigNumber) => {
+    return cGold.multipliedBy(cDollarToCGoldRate[0]).dividedBy(cDollarToCGoldRate[1])
+  }
+
+  describe('Tx pool tests', () => {
+    it('when tx pool gets full, ensure that celo gold gas denominated tx gets pruned', async () => {
+      const fullInstance = {
+        name: 'txFull',
+        validating: false,
+        syncmode: 'full',
+        lightserv: true,
+        port: 30305,
+        rpcport: 8545,
+        maxtxpoolsize: 1,
+        privateKey: DEF_FROM_PK,
+      }
+      await restartGeth()
+      await startGeth(hooks.gethBinaryPath, fullInstance)
+      await sleep(2)
+
+      web3 = new Web3('http://localhost:8545')
+
+      await unlockAccount(DEF_FROM_ADDR)
+
+      // Submit 3 transactions to 3 different addresses.  The second one has the lowest gas price, and should be the one that is pruned from the txpool.
```

Comment:
For my own understanding, why should it get pruned if it has a different nonce?

Reply:
I'll post a summary of how the txpool works in the #geth-questions slack channel.  The details of it actually is not very intuitive and has alot of nuances.